src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
+        Assertions.assertTrue(listResp.isEmpty());
+        CountDownLatch latch = new CountDownLatch(2);
+
+        // Create two new recordings
+        CompletableFuture<Void> createRecordingFuture1 = new CompletableFuture<>();
+        MultiMap form1 = MultiMap.caseInsensitiveMultiMap();
+        form1.add("recordingName", "Recording1");
+        form1.add("duration", "5");
+        form1.add("events", "template=ALL");
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form1,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture1)) {
+                                createRecordingFuture1.complete(null);
+                                latch.countDown();
+                            }
+                        });
+        createRecordingFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        CompletableFuture<Void> createRecordingFuture2 = new CompletableFuture<>();
+        MultiMap form2 = MultiMap.caseInsensitiveMultiMap();
+        form2.add("recordingName", "Recording2");
+        form2.add("duration", "5");
+        form2.add("events", "template=ALL");
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form2,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture2)) {
+                                createRecordingFuture2.complete(null);
+                                latch.countDown();
+                            }
+                        });
+        latch.await(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        // Verify recordings created
+        CompletableFuture<JsonArray> listRespFuture2 = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture2)) {
+                                listRespFuture2.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        listResp = listRespFuture2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(
+                "list should have size 2 after recording creation",
+                listResp.size(),
+                Matchers.equalTo(2));
+
+        // GraphQL Query to filter Active recordings by names
+        CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
+        String query =
+                "query { targetNodes {"
+                        + "recordings {"
+                        + "active(filter: { names: [\"Recording1\", \"Recording2\"] }) {"
```

Comment:
If two recordings are created, and this filters for both of their expected values, then we don't know if this filter actually does anything - it might just pass through all results. Should this test perhaps query for only one of them? Or query with a filter that matches neither and assert that the response contains no entries? etc.

Reply:
added one extra that will not be filtered